consul_exporter.go

```diff
@@ -213,21 +213,22 @@ func (e *Exporter) Collect(ch chan<- prometheus.Metric) {
 	leader, err := e.client.Status().Leader()
 	if err != nil {
 		log.Errorf("Can't query consul: %v", err)
-	}
-	if len(leader) == 0 {
-		ch <- prometheus.MustNewConstMetric(
-			clusterLeader, prometheus.GaugeValue, 0,
-		)
 	} else {
-		ch <- prometheus.MustNewConstMetric(
-			clusterLeader, prometheus.GaugeValue, 1,
-		)
+		if len(leader) == 0 {
```

Comment:
You can't use the return value if err != nill.

Reply:
Ah, good point. I'll update the specific `err` variable names so it's not shadowed throughout this function.